### PR TITLE
Ability to ignore packages 

### LINKED ISSUE #41: Ignore packages
One should be able to ignore packages.

Lets say test is not able to resolve certain license. However, I am able to find online its license and confirm that it is ok for my scenario. I should be able to add it to config file to prevent further fails.

```
ignore: [
'package-utils@1.0.0',
'random@1.2.3'
]
```

### PR DESCRIPTION
**Pull request by jvorcak**
Fixes #41 

You can now optionally specify this in your config
```
"ignored_packages": {
    "python_pipenv": [
      "PyInquirer",
      "setuptools"
    ],
    "npm": [],
    "maven": [],
    "yarn": []
  }
```

### CODE REVIEW

File: license_sh/helpers.py [hunk 1]
```diff
@@ -92,7 +95,10 @@ def annotate_dep_tree(tree, whitelist: [str]) -> Tuple[AnyNode, Set[str]]:
 
     licenses_not_found = set()
     for node in PreOrderIter(tree):
-        node.license_problem = not is_license_ok(node.license_normalized, whitelist)
+        node.license_problem = (
+            not is_license_ok(node.license_normalized, whitelist)
+            and node.name not in ignored_packages
```
Comment (S4n60w3n) [inline license_sh/helpers.py]: Node.name concist of name and version? If not this is wrong

File: tests/runners/test_helpers.py [hunk 2]
```diff
@@ -73,14 +73,14 @@ def test_bad_licenses_identified(self):
         tree = get_tree()
 
         whitelist = ["MIT", "Apache-2.0"]
-        _, unknown_licenses = annotate_dep_tree(tree, whitelist)
+        _, unknown_licenses = annotate_dep_tree(tree, whitelist, [])
```
Comment (S4n60w3n) [inline tests/runners/test_helpers.py]: This is hardly a test for this functionality

File: license-sh [hunk 3]
```diff
@@ -84,21 +91,25 @@ if __name__ == "__main__":
     if ProjectType.PYTHON_PIPENV in project_types:
         runner = PythonRunner(path, silent, debug)
         dep_tree, license_map = runner.check()
+        ignored_packages = ignored_packages_map[ProjectType.PYTHON_PIPENV.value]
 
     if ProjectType.NPM in project_types:
         runner = NpmRunner(path, silent, debug)
         dep_tree, license_map = runner.check()
+        ignored_packages = ignored_packages_map[ProjectType.PYTHON_PIPENV.value]
```
Comment (S4n60w3n) [inline license-sh]: PIPENV?

File: license-sh [hunk 4]
```diff
@@ -84,21 +91,25 @@ if __name__ == "__main__":
...
 
     if ProjectType.MAVEN in project_types:
         runner = MavenRunner(path, silent, debug)
         dep_tree, license_map = runner.check()
+        ignored_packages = ignored_packages_map[ProjectType.PYTHON_PIPENV.value]
```
Comment (S4n60w3n) [inline license-sh]: PIPENV?

File: license-sh [hunk 5]
```diff
@@ -84,21 +91,25 @@ if __name__ == "__main__":
...
 
     if ProjectType.YARN in project_types:
         runner = YarnRunner(path, silent, debug)
         dep_tree, license_map = runner.check()
+        ignored_packages = ignored_packages_map[ProjectType.PYTHON_PIPENV.value]
```
Comment (S4n60w3n) [inline license-sh]: PIPENV?

File: license-sh (hunk 3 above)
Comment (jvorcak) [reply license-sh]: https://github.com/pypa/pipenv @S4n60w3n I don't get this comment

File: license_sh/helpers.py (hunk 1 above)
Comment (jvorcak) [reply license_sh/helpers.py]: @S4n60w3n as I wrote here https://github.com/webscopeio/license.sh/issues/41#issuecomment-566527158, we will support only names in the config for now. 

Each language has different versioning mechanism, so I'd start with a name and continue with supporting semver later.

File: tests/runners/test_helpers.py (hunk 2 above)
Comment (jvorcak) [reply tests/runners/test_helpers.py]: This is just to make this test backwards compatible. I'll add a test for this functionality in a different test case. 👍

**Comment by jvorcak** [comment]
I've added test cases. This is quite a simple functionality so I couldn't think of more corner cases.
